OpenRA.Mods.Common/ServerTraits/MasterServerPinger.cs

```diff
@@ -23,34 +24,48 @@ public class MasterServerPinger : ServerTrait, ITick, INotifySyncLobbyInfo, ISta
 	{
 		// 3 minutes. Server has a 5 minute TTL for games, so give ourselves a bit of leeway.
 		const int MasterPingInterval = 60 * 3;
+		public const string OpenRALANGame = "OpenRALANGame";
+		readonly int currentProcessId = System.Diagnostics.Process.GetCurrentProcess().Id;
+
 		public int TickTimeout { get { return MasterPingInterval * 10000; } }
 
+		Beacon lanGameBeacon;
+		string lastLanGameBeacon = "";
+
 		public void Tick(S server)
 		{
 			if ((Game.RunTime - lastPing > MasterPingInterval * 1000) || isInitialPing)
-				PingMasterServer(server);
+				PingMasterServer(server, false);
 			else
 				lock (masterServerMessages)
 					while (masterServerMessages.Count > 0)
 						server.SendMessage(masterServerMessages.Dequeue());
 		}
 
-		public void LobbyInfoSynced(S server) { PingMasterServer(server); }
-		public void GameStarted(S server) { PingMasterServer(server); }
-		public void GameEnded(S server) { PingMasterServer(server); }
+		public void LobbyInfoSynced(S server) { PingMasterServer(server, true); }
+		public void GameStarted(S server) { PingMasterServer(server, true); }
+
+		public void GameEnded(S server)
+		{
+			if (lanGameBeacon != null)
+			{
+				lanGameBeacon.Stop();
+				lastLanGameBeacon = "";
+			}
+
+			PingMasterServer(server, false);
+		}
 
 		long lastPing = 0;
 		bool isInitialPing = true;
 
 		volatile bool isBusy;
 		Queue<string> masterServerMessages = new Queue<string>();
 
-		public void PingMasterServer(S server)
+		public void PingMasterServer(S server, bool createLANGameBeacon)
```

Comment:
This method now feels a bit too spaghettiish.   I suggest pulling the `numPlayers` etc out into another method that then calls this and `CreateLANGameBeacon` individually.